src/OrchardCore.Modules/OrchardCore.Users/Startup.cs

```diff
@@ -139,6 +139,7 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddLiquidFilter<HasPermissionFilter>("has_permission");
             services.AddLiquidFilter<HasClaimFilter>("has_claim");
             services.AddLiquidFilter<IsInRoleFilter>("is_in_role");
+            TemplateContext.GlobalMemberAccessStrategy.Register<User>();
         }
```

Comment:
Not sure about security concerns

but would need to be done in a static ctor as done in other modules, e.g

        static Startup()
        {
            TemplateContext.GlobalMemberAccessStrategy.Register<AutoroutePartViewModel>();
        }

But i think better to add it in the already existing `UserLiquidTemplateEventHandler`.


        public Task RenderingAsync(TemplateContext context)
        {
            var user = _httpContextAccessor.HttpContext.User;

            context.MemberAccessStrategy.Register<User>();
            context.MemberAccessStrategy.Register<ClaimsPrincipal>();
            context.MemberAccessStrategy.Register<ClaimsIdentity>();
            context.SetValue("User", user);

            return Task.CompletedTask;
        }

Reply:
I will try

Reply:
Not sure why it would not be secure to register it globally. One thing comes to my mind here and it's the fact that I could want to also access my own IUser extension.

Reply:
It works for the Workflow that way so you can push a PR since you found the solution :)

Reply:
Okay but for thread safety you need to add it on the global `TemplateContext` in a static constructor of the startup class.

But better to add it on the scoped `context` in the event handler (in the same module) that already add some access strategies, **just one line to add in `UserLiquidTemplateEventHandler.cs`**.

Reply:
Is it not redundant to allow it in each scope for a Liquid TemplateContext? If the intent is to avoid security issues we still register it in each scope so it's available as a Liquid accessible type everytime we invoke the UserLiquidTemplateEventHandler.RenderAsync task. Here the difference is that you register it everytime the RenderingAsync is called. So, in the end the result is the same. I don't know to be honest. I need some sleep 😉

Reply:
No problem, it's just 2 ways to register it

But doing it as you did, not in a static ctor, is not thread safe because the global `TemplateContext.GlobalMemberAccessStrategy` **is a static property** that may be updated by multiple tenants doing their startups concurrently. I talk about concurrent issues, not about security.

Just saw that `Fluid` now uses a `ConcurrentMemberAccessStrategy`, not sure but i think this was not the case before. But still better for a given module startup to do it once in a static ctor and for all tenants.

Then i suggested to do it in the handler of the same module just because it was already doing it for other types. But i agree, seems to be better to do it once in the startup, and maybe move all of them, but if so in a static ctor.

I don't remember why we use these 2 ways, at least we need this scope level handler for doing things as context.SetValue("User", user).

**Oh yes, because if you disable a module / feature, its related access strategies would not be added anymore, but if they have been added globaly they would be always there**.

We will see tomorrow ;)

Reply:
> Oh yes, because if you disable a module / feature, its related access strategies would not be added anymore, but if they have been added globaly they would be always there.

But here we register it as part of the Liquid Feature of that User module so it should activate only when the Liquid module is enabled?

Reply:
Because a `ConcurrentMemberAccessStrategy` [has been added here](https://github.com/sebastienros/fluid/commit/06eb2aabb09312099ecb7b6d64e81a27d459a667#diff-b400a916eec233c138f6a83cce513954) maybe there is no more concurrency issues but `TemplateContext.GlobalMemberAccessStrategy` [is still a global static object](https://github.com/sebastienros/fluid/blob/dev/Fluid/TemplateContext.cs#L70). So as soon as you will enable the feature the strategy will be registered, but the strategy will be always there even if you disable the feature.

So here the right place is the event handler, this to add the strategy through a scoped context.